### PR TITLE
wipes profile section when bmx write --use-credential-process is ran

### PR DESCRIPTION
**Pull request by jkomonen**
### Why
If there's a profile with the same name in the credentials and config file, it will default to the credentials file, so you will eventually get invalid credentials

### Ticket
https://desire2learn.atlassian.net/browse/VUL-329

### CODE REVIEW

File: src/D2L.Bmx/WriteHandler.cs [hunk 1]
```diff
@@ -70,6 +70,11 @@ bool useCredentialProcess
 			if( !data.Sections.ContainsSection( sectionName ) ) {
 				data.Sections.AddSection( sectionName );
 			}
+			var defaultCredentialsFile = parser.ReadFile( SharedCredentialsFile.DefaultFilePath );
```
Comment (cfbao) [inline src/D2L.Bmx/WriteHandler.cs:73]: We should probably check that a file exists at `SharedCredentialsFile.DefaultFilePath` first.

File: src/D2L.Bmx/WriteHandler.cs [hunk 2]
```diff
@@ -70,6 +70,11 @@ bool useCredentialProcess
...
+			if (defaultCredentialsFile.Sections.ContainsSection( profile ) ) {
+				defaultCredentialsFile.Sections.RemoveSection( profile );
+				parser.WriteFile(SharedCredentialsFile.DefaultFilePath, defaultCredentialsFile);
```
Comment (cfbao) [inline src/D2L.Bmx/WriteHandler.cs:76]: Formatting looks off here.
Please check the auto formatting setup in your IDE and make sure it respects editorconfig. Let us know if you need any help with that.

**Review by cfbao — COMMENTED** [review]
Code change looks good, but please merge the auto format PR to fix the formatting.
Still need to look into your IDE's auto format setup.

**Comment by jkomonen** [comment]
> Code change looks good, but please merge the auto format PR to fix the formatting. Still need to look into your IDE's auto format setup.

Where can I find the documentation to set this up?

**Comment by cfbao** [comment]
Here: https://desire2learn.atlassian.net/wiki/spaces/MaT/pages/1470857332/IDE+setup+and+auto+formatting
It's mentioned in the "set up dev machine" onboarding card.
If you set this up but it isn't working as expected, please reach out.

---

Aside, please @ people in GitHub when you're "replying" on the main PR conversation thread, because otherwise we won't get a direct notification.
(Replying to review comments doesn't require @'ing people. We always get direct notification for these.)

**Review by cfbao — APPROVED** [review]

**Pull request merged**